### ISSUE TITLE
the very much needed width / height

### ISSUE THREAD

**Issue by delahee** [ncannasse/heaps#114]
As time goes we forget the littlest problems, but this one is big, very big.

The width/height notion is missing and very much needed for anyone wanting to do 2d things seriously.

We had it in h3d (https://github.com/motion-twin/h3d/blob/master/h2d/Sprite.hx), but as it was not something designed from the root by Nicolas, there was mishaps  here and there and it was literally unnerving not to have such a simple thing to work.

Basically for Bitmap it was working this way : 
```
@:isVar
public var width(get, set) : Float;
@:isVar
public var height(get, set) : Float;
public function get_width():Float { 
		var b = Tools.getCoreObjects().tmpBounds;
		b.empty();
		return getBounds(parent , b ).width;
	}
	
	public function get_height():Float { 
		var b = Tools.getCoreObjects().tmpBounds;
		b.empty();
		return getBounds(parent,b).height;
	}
```

I know it s a long tedious refactor for you, I can send a patch as the work was done cahin caha BUT since it touches nearly all classes of h2d, there would be numerous points of friction that would need discussion and ultimately, this would result us both wasting more time than just you going throught the whole thing and adding a definitive "width" and "height" definition in heaps.

if you want less duty, maybe naming width/height to content width/Height might better suit you but ultimately, I don't thing anyone but you can do it seeing the numerous classes that are touched.

Thansk for reading, keep me posted.

**Comment by ncannasse** [comment]
We have getBounds() already no ? so why not adding it this way ? (PS : you don't want/need @:isVar here)

**Comment by ncannasse** [comment]
PS : the real problem is about performances, since each access might make a lot of calculus

**Comment by delahee** [comment]
well we throgoughly tested the approach, the performance issue is not really here as its often on resize calls the burden is and frankly even on mobile it was nearly peanut. 

The getBounds fallback is...well there are badly handled  in many many cases (scale on text etc...) and it is really not good at ergonomics. It just make the 2d code ugly and hard to follow.

You should talk aabout this with @deepnight too :)

**Comment by ncannasse** [comment]
There's not much discussions needed here :)
- we can add width/height based on getBounds results (general case)
- we can override width/height for Bitmap and a few others to use tile size instead
- any issue wrt getBounds calculation should be reported and fixed, since it's well specified (it should give the bounds in the space of relativeTo, or the scene if not specified)

**Comment by delahee** [comment]
Cool let's start with 1!

**Comment by ncannasse** [comment]
I've added getSize() on Sprite, which returns an h2d.col.Bounds.
Please note that for h2d.Text/HtmlText (and Flow if isInline=true) it's slightly different than getBounds since it will return the height based on the text baseLine instead.

**Comment by delahee** [comment]
After reading, seems awesome to me and much better than what I would have done :). Thanks.
